src/validation.cpp

```diff
@@ -1485,6 +1485,7 @@ int ApplyTxInUndo(Coin&& undo, CCoinsViewCache& view, const COutPoint& out)
  *  When FAILED is returned, view is left in an indeterminate state. */
 DisconnectResult CChainState::DisconnectBlock(const CBlock& block, const CBlockIndex* pindex, CCoinsViewCache& view)
 {
+    AssertLockHeld(::cs_main);
```

Comment:
This assertion in the definition seems unneeded? IIUC the lock annotation in the function declaration enforces it.

Reply:
Never mind, just saw the developer notes recommend this ("Combine annotations in function declarations with
  run-time asserts in function definitions").